compiler/rustc_trait_selection/src/traits/specialize/mod.rs

```diff
@@ -230,23 +231,26 @@ pub(super) fn specialization_enabled_in(tcx: TyCtxt<'_>, _: LocalCrate) -> bool
 /// `impl1` specializes `impl2` if it applies to a subset of the types `impl2` applies
```

Comment:
also, I feel like it's very helpful to explicitly write down "We therefore prove that the specializing impl *implies* the parent impl".

Because then it feels clear why we're instantiating the specializing impl with placeholders and the parent impl with existentials and prove all bounds of the parent impül